### PR TITLE
[thethingsnetwork] Upgrade to TheThingsNetwork v3

### PR DESCRIPTION
**Pull request by angelnu**
## Proposed change

Update documentation to match the updated thethingsnetwork integration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/113375
- Link to parent pull request in the Brands repository: NA
- This PR fixes or closes issue: NA

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Transitioned The Things Network integration from TTNv2 to TTNv3, enhancing compatibility and future-proofing.

- **Improvements**
  - Updated integration links to the latest documentation.
  - Refined setup instructions for easier configuration.

- **Changes**
  - Changed IoT class from Local Push to Cloud Polling for better performance.
  - Updated code ownership to ensure better maintenance and support.

### CODE REVIEW

**Comment by angelnu** [comment]
This is a re-submittion of the previously merged https://github.com/home-assistant/home-assistant.io/pull/31930 which was reverted.

**Comment by joostlek** [comment]
This should be rebased on next

**Comment by angelnu** [comment]
Changing the target branch did not work as expecting - closing and re-opening PR correctly

**Comment by angelnu** [comment]
Rebased on the next branch

File: source/_integrations/thethingsnetwork.markdown [hunk 1]
```diff
@@ -5,32 +5,28 @@ ha_category:
...
 ha_platforms:
   - sensor
 ha_integration_type: integration
 ---
 
-<div class='note warning'>
-  This integration only supports TTNv2 that has been definitively switched off in 2022 (deprecated).
-  For TTNv3, use the [MQTT integration](/integrations/mqtt).
-</div>
+The `thethingsnetwork` integration allows one to interact with the [The Things Network](https://www.thethingsnetwork.org). This community-driven and open network supports [LoRaWAN](https://www.lora-alliance.org/) for long range (~5 to 15 km) communication with a low bandwidth (51 bytes/message). [Gateways](https://www.thethingsnetwork.org/docs/gateways/) transfers the received data from the sensors to The Things Network.
```
Comment (c0ffeeca7) [inline source/_integrations/thethingsnetwork.markdown]:
```suggestion
The `thethingsnetwork` {% term integration%} allows you to interact with the [The Things Network](https://www.thethingsnetwork.org) from within Home Assistant. This community-driven and open network supports [LoRaWAN](https://www.lora-alliance.org/) for long-range (~5 to 15 km) communication with low bandwidth (51 bytes/message). [Gateways](https://www.thethingsnetwork.org/docs/gateways/) transfer the received data from the sensors to The Things Network.
```

File: source/_integrations/thethingsnetwork.markdown [hunk 2]
```diff
@@ -5,32 +5,28 @@ ha_category:
...
-| [MQTT](https://www.thethingsnetwork.org/docs/applications/mqtt/) | |
-| [Storage](https://www.thethingsnetwork.org/docs/applications/storage/) | [`thethingsnetwork`](#sensor) |
-| [HTTP](https://www.thethingsnetwork.org/docs/applications/http/) | |
+| [MQTT](https://www.thethingsindustries.com/docs/integrations/mqtt) | [`MQTT`](integrations/mqtt) |
+| [Storage](https://www.thethingsindustries.com/docs/integrations/storage) | [`thethingsnetwork`](#setup) |
+| [HTTP](https://www.thethingsindustries.com/docs/integrations/webhooks) | |
 
 There is currently support for the following device types within Home Assistant:
 
+- [Setup](#setup)
```
Comment (c0ffeeca7) [inline source/_integrations/thethingsnetwork.markdown]:
```suggestion
```

File: source/_integrations/thethingsnetwork.markdown [hunk 3]
```diff
@@ -5,32 +5,28 @@ ha_category:
...
 - [Sensor](#sensor)
 
 ## Setup
```
Comment (c0ffeeca7) [inline source/_integrations/thethingsnetwork.markdown]:
```suggestion
## Prerequisites
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

**Comment by c0ffeeca7** [comment]
I think "Visit the [The Things Network Console]...." should be the first step of the *Prerequistes* section.

**Comment by angelnu** [comment]
Note - the revert commit was taken out so most of this PR is already merged - I will need to to rebase on top of it so we do not loose the new feedback.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @angelnu :+1:

**Pull request merged**